### DISCUSSION TITLE
ESP8266 possible?

### DISCUSSION THREAD

**Discussion by mhos** [PlummersSoftwareLLC/NightDriverStrip#60] [General]
Hello,

Total noob to this type of project but have a background in IT and pretty good with figuring things out and reading and understanding code however, I am just starting out with any Arduino type project, did I mention noob, and this project peaked my interest from the second I had seen the video on youtube.
I have a bunch of ESP8266 mods that I had gotten cheap off amazon and waiting on my ESP32 to show up as it seems to me the code only works on that type of controller.
Am I wrong to assume that? Not knowledgeable enough yet to know the difference and couldn't find any instructions for this project on ESP8266 from my many searches online and here.

Was able to successfully flash WLED to one of them after fumbling with bad tool after bad tool that would not push the package. Might just be me, might be the board, IDK but bouncing back and forth from Ubuntu and Windows I finally got things working for WLED, its cool but really want to setup NightDriverStrip for my house eves lights. Already have some old regular led strips up there I am swapping out with addressable ones and would love to make this work on them.

Thanks for the read and any points, tips, examples anyone has for me starting out in this area of things, please send them my way.

**Comment by 9Rune5** [comment]
You could probably use the FastLED library directly (if NightDriverStrip doesn't simply work as-is).

I have a small project where I used NeoPixel, and that particular library can only drive one led strip on the ESP8266. So the ESP8266 seems to have a number of limitations and the path of least resistance is simply migrating to the ESP-32.

Another thing I only recently realized: The ESP-32 S2 is not necessarily better than the original ESP-32. (e.g. only a single core present). There is a discussion thread over at FastLed's github repo. (I bought three S2s before it dawned on me)

**Reply by mhos** [reply]
Thanks for the input. Yeah I kind of jumped the gun on the ESP8266 chips. They were cheap and got one per each strip and just assumed it would work with NightDriverStrip first. OOPS, but still can find a use for some smaller project im sure.
As for the ESP-32, yes I did catch the dont get the S2 model and I have the one recommended form amz on the readme of this project on the way after actually sitting down and going through it all.
Guess will have to see how it goes once the 32 shows up.
Thanks again for your response!

**Comment by huggy-d1** [comment]
Not a NightDriverStrip thing, but WLED allows an ESP8266 to control multiple strips with 2 output pins (possibly more pins for shorter LED strips and low fps).  WLED has to resort to bit-banging on some pins, so if you look in that direction for your ESP8266's, be sure and pick the recommended output pins.  If you hook up buttons, relays and such, try to stick with the default or suggested pin assignments as well.

**Comment by paul30003** [comment]
WLED works great on the ESP8266, that was my first led strip build. This project is far more feature packed though.
